core/src/main/scala-2/sttp/tapir/generic/auto/SchemaMagnoliaDerivation.scala

```diff
@@ -15,14 +15,14 @@ trait SchemaMagnoliaDerivation {
 
   def join[T](ctx: ReadOnlyCaseClass[Schema, T])(implicit genericDerivationConfig: Configuration): Schema[T] = {
     val annotations = mergeAnnotations(ctx.annotations, ctx.inheritedAnnotations)
-    withCache(ctx.typeName, annotations) {
```

Comment:
so ... we aren't using the merged annotations for caching and generating the schema name, but we *are* using them for other purposes? maybe add a comment to line 25 below that naming has to be *explicitly* declared as an annotation, and isn't inherited?

Reply:
and that's only for names of types, not for fields, right? :)

Reply:
Yes, we are using them for other purposes (`enrichSchema`). I added a comment and also explicitly moved the `mergedAnnotatations` call to `enrichSchema` for better clarity.
However, good question about field names. With my changes they will also stop inheriting encodedName. How can one generate such a case? Adding a `val` to the parent trait doesn't put a field on a case class anyway.

Reply:
hmm can't you have sth as follows?

```scala
sealed trait X {
  @encodedName("yy")
  val y: String
}

case class XX(y: String) extends X
```

I guess the encoded name should be inherited then

Reply:
OK, I tried it with `@description` but it wasn't working for at first, turns out I needed a `clean` :grin: 
I updated the PR to ensure it will work for `@encodedName` as well :)